appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java

```diff
@@ -7,48 +7,96 @@
 import android.content.Intent;
 import android.content.pm.PackageManager;
 import android.net.Uri;
+import android.util.ArrayMap;
+import android.util.Log;
 import com.asf.appcoins.sdk.ads.BuildConfig;
 import com.asf.appcoins.sdk.ads.R;
+import java.lang.reflect.Field;
 
-public class WalletUtils{
+public class WalletUtils {
 
-  private static String walletPackageName = BuildConfig.BDS_WALLET_PACKAGE_NAME;
+  private static boolean dialogVisible;
 
-  public static boolean hasWalletInstalled(Context context) {
+  public static String walletPackageName = BuildConfig.BDS_WALLET_PACKAGE_NAME;
+
+  public static Context context;
+
+  public static boolean isDialogVisible() {
```

Comment:
It should not be the WalletUtils to control if the dialog is already being shown.